### PR TITLE
GitHub Actions

### PR DESCRIPTION
**Pull request by justing-bq**
Defined github actions for build, goimports, and lint.

Also renamed mockError to errMock to satisfy the linter.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Set up Go 1.13
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.13
```
Comment (byronlin-bq) [inline .github/workflows/go.yml:13]: Please test with different go versions

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            dep ensure
+        fi
+    - name: Build
+      run: go build -v ./...
+    
+    - name: Test with coverage
+      run: go test -v ./... -coverprofile coverage.txt
+      
+      
+    - name: Upload Coverage report to CodeCov
```
Comment (byronlin-bq) [inline .github/workflows/go.yml:44]: Do we need this?

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        if [ -f Gopkg.toml ]; then
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+            dep ensure
+        fi
+    - name: Build
+      run: go build -v ./...
+    
+    - name: Test with coverage
+      run: go test -v ./... -coverprofile coverage.txt
+
```
Comment (byronlin-bq) [inline .github/workflows/go.yml:42]: Need to run integration test as well:

```
go test -run Integration
```

File: .github/workflows/go.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
+name: Go Build
+on: [push, pull_request]
```
Comment (byronlin-bq) [inline .github/workflows/go.yml:2]: Need to also add for push/pullrequest to master branch

File: .github/workflows/go.yml (hunk 1 above)
Comment (justing-bq) [reply .github/workflows/go.yml]: Done.

File: .github/workflows/go.yml (hunk 2 above)
Comment (justing-bq) [reply .github/workflows/go.yml]: Removed.

File: .github/workflows/go.yml (hunk 3 above)
Comment (justing-bq) [reply .github/workflows/go.yml]: Integration tests are already covered by `go test`

File: .github/workflows/go.yml (hunk 4 above)
Comment (justing-bq) [reply .github/workflows/go.yml]: Done.

File: .github/workflows/go.yml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    strategy:
```
Comment (Armanbqt) [inline .github/workflows/go.yml:13]: try ```max-parallel: 1``` after strategy instead of ledegrname+random number

File: .github/workflows/go.yml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+      matrix:
+        go_version: ['1.13', '1.14', '1.15']
+
+    steps:
+
+    - name: Set up Go version
+      uses: actions/setup-go@v1
```
Comment (Armanbqt) [inline .github/workflows/go.yml:20]: I noticed there is a newer version of ```setup-go```. Let's use v2

File: .github/workflows/go.yml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+            dep ensure
+        fi
+    - name: Build
+      run: go build -v ./...
+    
+    - name: Tests
```
Comment (byronlin-bq) [inline .github/workflows/go.yml:48]: Specify unit and integration tests

File: .github/workflows/go.yml (hunk 5 above)
Comment (justing-bq) [reply .github/workflows/go.yml:13]: Done.

File: .github/workflows/go.yml (hunk 6 above)
Comment (justing-bq) [reply .github/workflows/go.yml]: Done.

File: .github/workflows/go.yml (hunk 7 above)
Comment (justing-bq) [reply .github/workflows/go.yml]: Done.

File: .github/workflows/go.yml [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 1
+      matrix:
```
Comment (aurghob33) [inline .github/workflows/go.yml:15]: Can we add multiple os (os: [ubuntu-latest, macos-latest, windows-latest]) please?
https://github.com/aurghob33/amazon-qldb-driver-java/blob/master/.github/workflows/maven.yml has an example for java.

Comment (justing-bq) [reply .github/workflows/go.yml:15]: Done.

**Review by aurghob33 — APPROVED** [review]
Approving as build won't work in current PR as it is coming from a forked repo as per https://github.community/t/will-github-actions-support-pull-request-events-from-a-fork-to-a-private-base-repository/17471/59. Confirmed that build works in forked repo : https://github.com/justing-bq/amazon-qldb-driver-go/actions/runs/416277649 . Also confirmed that build works after change is approved in nodejs repository.

**Pull request merged**